src/store/gateway/ethereum.ts

```diff
@@ -538,11 +538,16 @@ export async function refreshEthereumHistory(context: ActionContext) {
 
 async function logEvents(address, gateway, symbol, depositEvent, withdrawEvent) {
   const cached = ethereumModule.state.history
-  const fromBlock = cached.length ? cached[0].blockNumber : 0
-  const range = {
-    fromBlock,
-    toBlock: "latest",
+  let fromBlock = cached.length ? cached[0].blockNumber : 0
+  let toBlock: string | number = "latest"
+  // BSC max query size is 5000
+  if (ethereumModule.state.walletType === "binance") {
```

Comment:
Should work based on the network id/name rather than the wallet type

Reply:
changed to `ethereumModule.state.networkName.startsWith("bsc-")`